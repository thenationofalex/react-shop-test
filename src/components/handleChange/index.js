export default function handleChange (e) {
  const target = e.target
  const value = target.value
  const name = target.id
  console.log(name, value)
  this.setState({
    [name]: value.toString()
  })
}
