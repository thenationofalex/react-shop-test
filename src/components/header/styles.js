import styled from 'react-emotion'

export const StyledHeader = styled('header')`
  background-color: ${props => props.theme.colors.blue};
  height: 3.35em;
  padding: 0 1.25em;
  position: relative;
  margin-bottom: 1.25em;
  & > h1 {
    margin: 0;
    line-height: 1.5em;
  }
`
