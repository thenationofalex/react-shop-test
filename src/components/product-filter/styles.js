import styled from 'react-emotion'

export const StyledSelect = styled('select')`
  position: absolute;
  top: 1.5em;
  right: 1.25em;
`
