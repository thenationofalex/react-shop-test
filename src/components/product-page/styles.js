import styled from 'react-emotion'

export const StyledPageWrapper = styled('div')`
  max-width: 1320px;
  margin: 0 auto;
`

export const StyledProductWrapper = styled('div')`
  display: flex;
  flex-direction: row;
  flex-flow: wrap;
  & > div {
    width: calc(100% - 2px);
  }
  @media (min-width: 600px) {
    & > div {
      width: calc(50% - 2px);
    }
  }
  @media (min-width: 900px) {
    & > div {
      width: calc(33.33% - 2px);
    }
  }
  @media (min-width: 1200px) {
    & > div {
      width: calc(25% - 2px);
    }
  }
`
