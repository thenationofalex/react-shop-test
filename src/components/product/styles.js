import styled from 'react-emotion'

export const StyledLabel = styled('div')`
  color: #fff;
  font-size: 1em;
  height: 100%;
  max-width: 45%;
  text-align: center;
  text-transform: capitalize;
  line-height: 3.125em;
`

export const StyledProduct = styled('div')`
  padding: 1.25em;
  outline-color #e1e1e1 !important;
  outline: 1px solid;
  margin-top: 1px;
  margin-left: 1px;
  & img {
    width: 100%;
  }
`

export const StyledLabelBlock = styled('div')`
  height: 3.125em;
  display: block;
  vertical-align: middle;
  margin: 0.65em 0;
`

export const StyledSale = styled(StyledLabel)`
  background-color: ${props => props.theme.colors.red};
`

export const StyledExclusive = styled(StyledLabel)`
  background-color: ${props => props.theme.colors.green};
`

export const StyledProductInfo = styled('div')`
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  & h2 {
    font-size: 1.25rem;
    font-weight: 400;
    margin: 0;
  }
  & p {
    font-size: 2.25rem;
    margin: 0;
  }
`
