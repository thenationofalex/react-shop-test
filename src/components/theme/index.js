export const theme = {
  colors: {
   green: '#009900',
   red: '#ff0000',
   blue: '#def1f4'
  }
}