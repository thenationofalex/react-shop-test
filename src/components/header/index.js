import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import { StyledHeader } from './styles'

class Header extends PureComponent {
  render () {
    return (
      <StyledHeader>
        <h1>{ this.props.title }</h1>
        { this.props.children }
      </StyledHeader>
    )
  }
}

Header.propTypes = {
  title: PropTypes.string,
  children: PropTypes.any
}

export default Header
