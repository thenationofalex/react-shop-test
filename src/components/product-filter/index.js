import React, { Fragment, PureComponent } from 'react'
import PropTypes from 'prop-types'
import flattenDeep from 'lodash/flattenDeep'
import uniq from 'lodash/uniq'

import { StyledSelect } from './styles'

let filterOptions = []

class ProductFilter extends PureComponent {
  render () {
    const { options, onChange } = this.props
    options.map(p => filterOptions.push(p.size))
    filterOptions = uniq(flattenDeep(filterOptions))
    return (
      <Fragment>
        <StyledSelect id='size' onChange={onChange}>
          <option value='reset'>Filter by size</option>
          {
            filterOptions.map(o => <option key={o} value={o}>{o}</option>)
          }
        </StyledSelect>
      </Fragment>
    )
  }
}

ProductFilter.propTypes = {
  options: PropTypes.array,
  onChange: PropTypes.func
}

export default ProductFilter
