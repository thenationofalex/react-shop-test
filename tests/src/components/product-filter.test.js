import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { theme } from '../../../src/components/theme'

import ProductFilter from '../../../src/components/product-filter'
import { StyledSelect } from '../../../src/components/product-filter/styles'

const testOpts = [
  {
    'index': 0,
    'isSale': true,
    'isExclusive': false,
    'price': '$18.88',
    'productImage': 'product-1.jpg',
    'productName': 'Striped shirt',
    'size': ['XS', 'S', 'L', 'XL']
  }
]

const props = {
  theme
}

describe('Component - Product filter', () => {
  test('Renders', () => {
    const wrapper = shallow(<ProductFilter onChange={() => jest.fn()} options={testOpts} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  test('Styled Component - StyledSelect - Renders', () => {
    const wrapper = shallow(<StyledSelect {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
