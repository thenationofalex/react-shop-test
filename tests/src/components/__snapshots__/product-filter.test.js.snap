// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Product filter Renders 1`] = `
<Fragment>
  <Styled(select)
    id="size"
    onChange={[Function]}
  >
    <option
      value="reset"
    >
      Filter by size
    </option>
    <option
      key="XS"
      value="XS"
    >
      XS
    </option>
    <option
      key="S"
      value="S"
    >
      S
    </option>
    <option
      key="L"
      value="L"
    >
      L
    </option>
    <option
      key="XL"
      value="XL"
    >
      XL
    </option>
  </Styled(select)>
</Fragment>
`;

exports[`Component - Product filter Styled Component - StyledSelect - Renders 1`] = `
<select
  className="css-1gf4wyf"
/>
`;
